test/ffmpeg-qsv/encode/10bit/hevc.py

```diff
@@ -88,6 +88,58 @@ def test_r2r(self, case, gop, slices, qp, quality, profile):
     vars(self).setdefault("r2r", 5)
     self.encode()
 
+class icq(HEVC10EncoderTest):
+  def init(self, tspec, case, gop, slices, bframes, quality, profile,
+          ladepth = None, lowpower = 0):
+    self.caps = platform.get_caps("encode", "hevc_10")
+    vars(self).update(tspec[case].copy())
+    vars(self).update(
+      bframes   = bframes,
+      case      = case,
+      gop       = gop,
+      profile   = profile,
+      quality   = quality,
+      rcmode    = "icq",
+      slices    = slices,
+      ladepth   = ladepth,
+      lowpower  = lowpower,
+    )
+
+  @slash.requires(*platform.have_caps("encode", "hevc_10"))
+  @slash.requires(*have_ffmpeg_encoder("hevc_qsv"))
+  @slash.requires(*have_ffmpeg_decoder("hevc_qsv"))
+  @slash.parametrize(*gen_hevc_icq_parameters(spec, ['main10']))
+  def test(self, case, gop, slices, bframes, quality, profile):
+    self.init(spec, case, gop, slices, bframes, quality, profile)
+    self.encode()
+
+  @slash.requires(*platform.have_caps("encode", "hevc_10"))
+  @slash.requires(*have_ffmpeg_encoder("hevc_qsv"))
+  @slash.requires(*have_ffmpeg_decoder("hevc_qsv"))
+  @slash.parametrize(*gen_hevc_icq_la_parameters(spec, ['main10']))
+  def test_la(self, case, gop, slices, bframes, quality, profile, ladepth):
```

Comment:
The convention has been to define separate classes for non-lowpower vs. lowpower vs. look-ahead.  For example, `class vbr` vs. `class vbr_lp` vs. `class vbr_la`.  Each convention has its pros and cons.  But please don't diverge from the historical convention.

Reply:
Ok, I will follow the convention to change my code.